docs/proposals/node-allocatable.md

```diff
@@ -0,0 +1,190 @@
+[`SystemReserved`](#system-reserved), but with a different semantic meaning. While system reserved
+designates resources set asside for system components, Reserved designates resources set asside for
+non-system components running outside of Kubernetes (currently this is reported as all the processes
+lumped together in the `/system` raw container).
+
+#### Allocatable flag
+
+If the `--allocatable` flag is set on the kubelet, then its value overrides the
+[`Allocatable`](#allocatable) calculation and is always reported instead. Initially this value can
+only be set at Kubelet start-up time, but will [eventually](#future-work) be converted to an
+updatable Config API (per-kubelet) parameter. Its value is specified using the same format as the
+[system reserved](#system-reserved) flag.
+
+## Issues
+
+### System reservation is smaller than system component usage
+
+**Solution**: Do nothing (best effort). Let the system daemons overflow the reserved resources and
+hope for the best. If the node usage is less than Allocatable, there will be some room for
+overflow and the node should continue to function. If the node has been scheduled to capacity
+(worst-case scenario) it may enter an unstable state, which is the current behavior in this
+situation.
+
+In the [future](#future-work) we may set a parent cgroup for system components, with limits set
+according to `SystemReserved`.
+
+### Version discrepancy
+
+**API server / scheduler is not allocatable-resources aware:** If the Kubelet rejects a Pod but the
+  scheduler expects the Kubelet to accept it, the system could get stuck in an infinite loop
+  scheduling a Pod onto the node only to have Kubelet repeatedly reject it. To avoid this situation,
+  we will do a 2-stage rollout of `Allocatable`. In stage 1 (targeted for 1.2), `Allocatable` will
+  be reported by the Kubelet and the scheduler will be updated to use it, but Kubelet will continue
+  to do admission checks based on `Capacity` (same as today). In stage 2 of the rollout (targeted
+  for 1.3 or later), the Kubelet will start doing admission checks based on `Allocatable`.
+
+**API server expects `Allocatable` but does not receive it:** If the kubelet is older and does not
+  provide `Allocatable` in the `NodeStatus`, then `Allocatable` will be
+  [defaulted](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/defaults.go) to
+  `Capacity` (which will yield todays behavior of scheduling based on capacity).
+
+### 3rd party schedulers
+
+The community should be notified that an update to schedulers is recommended, but if a scheduler is
+not updated it falls under the above case of "scheduler is not allocatable-resources aware".
+
+## Future work
+
+- Convert kubelet flags to Config API - Cleaner API, better contract, etc. See
+  [#12245](https://github.com/kubernetes/kubernetes/issues/12245).
+- Support dynamic updates to reservations. This requires the above change to migrate flags to a
```

Comment:
Updated to clarify. I was referring to updating the user-set values (`--system-reserved`, `--reserved`, `--allocatable`). If Kubelet is managing Allocatable (`--allocatable` is not set), then it will dynamically update Allocatable as kernel usage varies (everything else should be constant).